src/lib/message/L2ToL1Message.ts

```diff
@@ -298,7 +303,7 @@ export class L2ToL1MessageReader extends L2ToL1Message {
     return this.l1BatchNumber
   }
 
-  protected async getSendProps(l2Provider: Provider) {
+  protected async getSendProps(l2Provider: Provider, requireConfirmed = true) {
```

Comment:
I think a better way to do this would be to always look for the `latestNodeCreated` in getSendProps, but also store an isConfirmed flag for the resulting sendRootHash. You can populate isConfirmed by checking that `latestConfirmed` > `nodeNum`

I'm also ok with the way you have it, I think this flow is already a bit complicated which is why I'm suggesting the change above. If you decided to stick with the `requireConfirmed` flow I would suggest we make it non optional - typescript optional params can be a bit annoying around ordering, and I think it's generally nicer to be explicit by making it non optional.

Reply:
made it always look for the `latestNodeCreated` and added the `sendRootConfirmed` field